core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -317,41 +391,49 @@ int getHandleIndex(int entryIndex) {
     LookUp lookUp(K key) {
         // binary search sorted part of key array to quickly find node to start search at
         // it finds previous-to-key so start with its next
-        int curr = getEntryField(binaryFind(key), OFFSET_NEXT);
-        int cmp;
+        int curr = getEntryField(binaryFind(key), OFFSET.NEXT);
+        int cmp = -1;
         // iterate until end of list (or key is found)
+
         while (curr != NONE) {
             // compare current item's key to searched key
             cmp = compare(readKey(curr), key);
             // if item's key is larger - we've exceeded our key
             // it's not in chunk - no need to search further
-            if (cmp > 0)
+            if (cmp > 0) {
                 return null;
-                // if keys are equal - we've found the item
+            }
+            // if keys are equal - we've found the item
             else if (cmp == 0) {
-                int hi = getEntryField(curr, OFFSET_HANDLE_INDEX);
-                if (hi < 0) return new LookUp(null, curr, -1);
-                Handle h = handles[hi];
-                if (h.isDeleted()) return new LookUp(null, curr, hi);
-                return new LookUp(h, curr, hi);
+                long valueReference = getValueReference(curr);
```

Comment:
Why not to use getValueSlice directly here?

Reply:
I see that you use them both. Why do you need them both? Slice for simplicity of usage and reference for any future CAS? If so please add a comment about that.

Reply:
The comment is more about LookUp class

Reply:
We need the slice so we are able to write off-heap, and to signal that the value reference points to a deleted value (in case the slice is null and the reference is valid).
The reference is needed for the CAS later (e.g., when removing we need the old reference).

I think in the new version of @eranmeir (#42), where we only CAS `null` to a valid value, and a valid value to `null`, the reference is not always necessary (it is still needed for remove).